src/mailmap.rs

```diff
@@ -0,0 +1,207 @@
+use crate::commit::Commit;
+use crate::repository::Repository;
+use crate::signature::{Signature, SignaturePayload};
+use napi::bindgen_prelude::*;
+use napi_derive::napi;
+use std::ops::{Deref, DerefMut};
+
+pub(crate) enum MailmapInner {
+  Repo(SharedReference<Repository, git2::Mailmap>),
+  Owned(git2::Mailmap),
+}
+
+impl Deref for MailmapInner {
+  type Target = git2::Mailmap;
+
+  fn deref(&self) -> &Self::Target {
+    match self {
+      Self::Repo(repo) => repo.deref(),
+      Self::Owned(mailmap) => mailmap,
+    }
+  }
+}
+
+impl DerefMut for MailmapInner {
+  fn deref_mut(&mut self) -> &mut Self::Target {
+    match self {
+      Self::Repo(repo) => repo.deref_mut(),
+      Self::Owned(mailmap) => mailmap,
+    }
+  }
+}
+
+/// A wrapper around git2::Mailmap providing Node.js bindings
+#[napi]
+pub struct Mailmap {
+  pub(crate) inner: MailmapInner,
+}
+
+#[napi]
+impl Mailmap {
+  /// Create a mailmap from the contents of a string.
+  ///
+  /// The format of the string should follow the rules of the mailmap file:
+  /// ```
+  /// # Comment line (ignored)
+  /// Seokju Me <seokju.me@toss.im> Seokju Na <seokju.me@gmail.com>
+  /// ```
+  ///
+  /// @param {string} content - Content of the mailmap file
+  /// @returns A new mailmap object or null if operation failed
+  #[napi]
+  pub fn from_buffer(content: String) -> crate::Result<Option<Mailmap>> {
```

Comment:
In this project, we prefer to use functions rather than static classes for the constructor. How about designing an interface with function named `createMailmapFromBuffer`?
In addition, since this function is a creation operation, I think it would be better to throw an error rather than return `null` if creation fails.

```suggestion
pub fn create_mailmap_from_buffer(content: String) -> crate::Result<Mailmap> {
  todo!()
}
```